### PR TITLE
Knockout/Weaken from melee weapons change (WIP)

### PR DESCRIPTION
**Pull request by yackemflam**
Changing it so that the weapon/item knockdown chance =/= force

### CODE REVIEW

**Comment by Ergovisavi** [comment]
Said it in coderbus and I'll say it here: Changing this doesn't change the fact that an RNG Knockout chance will never feel good on the receiving end.

**Comment by yackemflam** [comment]
Aye, but this willhelps steer us away from stun based combat.

**Comment by KorPhaeron** [comment]
What is the intended replacement?

**Comment by yackemflam** [comment]
It'll still be the same(ish).
All I'm doing is making the stun chance it's own variable.
So you can have a 40 force weapon with a 15% stun chance.

**Comment by optimumtact** [comment]
>Really, you guys have a really sloppy code that relies too much on it self.
Get a load of this guy! :laughing:

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -566,6 +566,23 @@ obj/item/proc/item_action_slot_check(slot, mob/user)
 		. = 'sound/weapons/sear.ogg'
 	else
 		. = pick('sound/misc/desceration-01.ogg', 'sound/misc/desceration-02.ogg', 'sound/misc/desceration-03.ogg')
+		
+	
+/obj/item/proc/get_knockout_chance()
+	switch(hit_area)
+		if("head")	//Harder to score a stun but if you do it lasts a bit longerswitch(hit_area)
+			if(H.stat == CONSCIOUS && armor_block < 50)
+				if(prob(Knockout))
```
Comment (optimumtact) [inline code/game/objects/items.dm]: knockout

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1071,10 +1070,6 @@
 		switch(hit_area)
 			if("head")	//Harder to score a stun but if you do it lasts a bit longer
 				if(H.stat == CONSCIOUS && armor_block < 50)
-					if(prob(I.force))
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/species.dm]: you probably want to call your new function here somewhere.

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -566,6 +566,23 @@ obj/item/proc/item_action_slot_check(slot, mob/user)
 		. = 'sound/weapons/sear.ogg'
 	else
 		. = pick('sound/misc/desceration-01.ogg', 'sound/misc/desceration-02.ogg', 'sound/misc/desceration-03.ogg')
+		
+	
+/obj/item/proc/get_knockout_chance()
+	switch(hit_area)
```
Comment (optimumtact) [inline code/game/objects/items.dm]: you need to pass through the H, I and hit_area variables to this proc or get them via some other means.

File: code/game/objects/items/weapons/weaponry.dm [hunk 4]
```diff
@@ -28,6 +28,7 @@
 	force = 2
 	throwforce = 1
 	w_class = 3
+	knockout = 100
```
Comment (optimumtact) [inline code/game/objects/items/weapons/weaponry.dm]: if this isn't already defined you need to define it on the /obj/item definition.

**Comment by Cheridan** [comment]
>another var for people to make balance meme prs about forever 

:disappointed:

File: code/game/objects/items/weapons/weaponry.dm (hunk 4 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/weaponry.dm]: >buffing the sord
>making it an objectively better stun baton

what the fuck

**Comment by PKPenguin321** [comment]
this is a really bad PR :-1:

File: code/game/objects/items/weapons/weaponry.dm (hunk 4 above)
Comment (optimumtact) [inline code/game/objects/items/weapons/weaponry.dm]: also yeah no.

File: code/game/objects/items.dm [hunk 5]
```diff
@@ -566,6 +566,23 @@ obj/item/proc/item_action_slot_check(slot, mob/user)
...
+		
+	
+/obj/item/proc/get_knockout_chance()
+	switch(hit_area)
+		if("head")	//Harder to score a stun but if you do it lasts a bit longerswitch(hit_area)
+			if(H.stat == CONSCIOUS && armor_block < 50)
+				if(prob(Knockout))
+					H.visible_message("<span class='danger'>[H] has been knocked unconscious!</span>", \
+						"<span class='userdanger'>[H] has been knocked unconscious!</span>")
+					H.apply_effect(20, PARALYZE, armor_block)
```
Comment (Iamgoofball) [inline code/game/objects/items.dm]: >20 seconds

File: code/game/objects/items.dm [hunk 6]
```diff
@@ -566,6 +566,23 @@ obj/item/proc/item_action_slot_check(slot, mob/user)
...
+
+		if("chest")	//Easier to score a stun but lasts less time
+			if(H.stat == CONSCIOUS && I.force && prob(Knockout + 10))
+				H.visible_message("<span class='danger'>[H] has been knocked down!</span>", \
+					"<span class='userdanger'>[H] has been knocked down!</span>")
+				H.apply_effect(5, WEAKEN, armor_block)
```
Comment (Iamgoofball) [inline code/game/objects/items.dm]: >stun baton

File: code/game/objects/items.dm (hunk 5 above)
Comment (Fox-McCloud) [inline code/game/objects/items.dm]: This is actually on live, Goof.

It needs to go, IMO.

File: code/game/objects/items/weapons/weaponry.dm (hunk 4 above)
Comment (KorPhaeron) [inline code/game/objects/items/weapons/weaponry.dm]: why is the concept of atomizing prs so hard for people to grasp

**Comment by PKPenguin321** [comment]
this is a pretty garbage PR and im surprised it hasnt already been shut down

**Comment by Niknakflak** [comment]
Bad

**Comment by HankMoodyExperience** [comment]
fucks sake just remove stuns entirely and be done with it

**Comment by yackemflam** [comment]
Holy shit people.
1. I just started
2. The SORD is for testing, I'll revert that later.
3. Shut the fuck up goof.

**Comment by yackemflam** [comment]
I got a better way to do this.

**Pull request closed without merging**